packages/story-editor/src/app/rightClickMenu/hooks/useCopyPasteActions.js

```diff
@@ -0,0 +1,238 @@
+import useApplyStyle from '../../../components/panels/design/textStyle/stylePresets/useApplyStyle';
+import getUpdatedSizeAndPosition from '../../../utils/getUpdatedSizeAndPosition';
+import { getTextPresets } from '../../../utils/presetUtils';
+import { useHistory, useStory } from '../..';
+import { getElementStyles } from './utils';
+import { UNDO_HELP_TEXT } from './constants';
+
+/**
+ * Creates the right click menu copy/paste actions.
+ *
+ * @return {Object} Right click menu copy/paste actions
+ */
+const useCopyPasteActions = () => {
+  const undo = useHistory(({ actions }) => actions.undo);
+  const {
+    addAnimations,
+    selectedElement,
+    selectedElementAnimations,
+    updateElementsById,
+  } = useStory(({ state, actions }) => ({
+    addAnimations: actions.addAnimations,
+    selectedElement: state.selectedElements?.[0],
+    selectedElementAnimations: state.selectedElementAnimations,
+    updateElementsById: actions.updateElementsById,
+  }));
+
+  const { showSnackbar } = useSnackbar();
+
+  // Needed to not pass stale refs of `undo` to snackbar
+  const undoRef = useRef(undo);
+  undoRef.current = undo;
+
+  const [copiedElement, setCopiedElement] = useState(null);
+
+  /**
+   * Copy the styles and animations of the selected element.
+   */
+  const handleCopyStyles = useCallback(() => {
+    const oldStyles = { ...copiedElement };
+
+    setCopiedElement({
+      animations: selectedElementAnimations,
+      styles: getElementStyles(selectedElement),
+      type: selectedElement?.type,
+    });
+
+    showSnackbar({
+      actionLabel: __('Undo', 'web-stories'),
+      dismissible: false,
+      message: __('Copied style.', 'web-stories'),
+      onAction: () => {
+        setCopiedElement(oldStyles);
+
+        trackEvent('context_menu_action', {
+          name: 'undo_copy_styles',
+          element: selectedElement?.type,
+          isBackground: selectedElement?.isBackground,
+        });
+      },
+      actionHelpText: UNDO_HELP_TEXT,
+    });
+
+    trackEvent('context_menu_action', {
+      name: 'copy_styles',
+      element: selectedElement?.type,
+      isBackground: selectedElement?.isBackground,
+    });
+  }, [copiedElement, selectedElement, selectedElementAnimations, showSnackbar]);
+
+  const selectedElementId = selectedElement?.id;
+  const pushUpdate = useCallback(
+    (update, commitValues) => {
+      if (selectedElementId) {
+        updateElementsById({
+          elementIds: [selectedElementId],
+          properties: (element) => {
+            const updates = updateProperties(element, update, commitValues);
+            const sizeUpdates = getUpdatedSizeAndPosition({
+              ...element,
+              ...updates,
+            });
+            return {
+              ...updates,
+              ...sizeUpdates,
+            };
+          },
+        });
+      }
+    },
+    [selectedElementId, updateElementsById]
+  );
+
+  const handleApplyStyle = useApplyStyle({ pushUpdate });
```

Comment:
This comment + exploration is amazing. Thank you for such a detailed dive. I'm going to focus on this in the following ticket. This will help avoid scope creep and make it easier to review those changes.